src/com/linkedin/parseq/Engine.java

```diff
@@ -204,6 +205,14 @@ public boolean awaitTermination(final int time, final TimeUnit unit) throws Inte
     return _terminated.await(time, unit);
   }
 
+  public boolean addPlanActivityListener(final PlanActivityListener planActivityListener) {
+    return _planActivityListeners.add(planActivityListener);
+  }
+
```

Comment:
Yeah, that's a tradeoff... I think it would be easier to reason/debug if the listeners are provided at construction time.  But if we don't have this use case, probably we can add the dynamic part later when there is a request/need, since it's backward compatible change.